ts/torch_handler/request_envelope/kservev2.py

```diff
@@ -87,20 +89,38 @@ def _batch_from_json(self, rows):
         Joins the instances of a batch of JSON objects
         """
         logger.debug("Parse input data %s", rows)
-        body_list = [body_list.get("data") or body_list.get("body") for body_list in rows]
+        body_list = [
+            body_list.get("data") or body_list.get("body") for body_list in rows
+        ]
         data_list = self._from_json(body_list)
         return data_list
 
     def _from_json(self, body_list):
         """
         Extracts the data from the JSON object
         """
-        # If the KF Transformer and Explainer sends in data as bytesarray
         if isinstance(body_list[0], (bytes, bytearray)):
-            body_list = [json.loads(body.decode()) for body in body_list]
+            body_list = [json.loads(body.decode("utf8")) for body in body_list]
             logger.debug("Bytes array is %s", body_list)
-        if "id" in body_list[0]:
+
+        input_names = []
+        for index, input in enumerate(body_list[0]["inputs"]):
+            if input["datatype"] == "BYTES":
+                body_list[0]["inputs"][index]["data"] = input["data"][0]
+            else:
+                body_list[0]["inputs"][index]["data"] = (
+                    np.array(input["data"]).reshape(tuple(input["shape"])).tolist()
```

Comment:
this PR and [PR](https://github.com/pytorch/serve/pull/2175) need sync up for this part.

Reply:
Yes added PR #2175 as dependent PR.